sql/sp.cc

```diff
@@ -2694,6 +2798,104 @@ Sp_handler::sp_resolve_package_routine(THD *thd,
 }
 
 
+/**
+  Detect cases when a package routine (rather than a standalone routine)
+  is called, and rewrite sp_name accordingly.
+  Note that this function is extends from the function,
+  Sp_handler::sp_resolve_package_routine()
+
+  @param thd              Current thd
+  @param caller           The caller routine (or NULL if outside of a routine)
+  @param [IN/OUT] name    The called routine name
+  @param [OUT]    pkgname If the routine is found to be a package routine,
+                          pkgname is populated with the package name.
+                          Otherwise, it's not touched.
+  @retval         false   on success
+  @retval         true    on error (e.g. EOM, could not read CREATE PACKAGE)
+*/
+
+bool
+Sp_handler::sp_resolve_package_routine_sql_path(THD *thd,
+                                       sp_head *caller,
+                                       sp_name *name,
+                                       const Sp_handler **pkg_routine_handler,
+                                       Database_qualified_name *pkgname) const
+{
+  bool ret= false;
+  sp_name *qname= NULL;
+
+  if (!thd->db.length)
+  {
+    if (name->m_explicit_name)
+      ret= thd->variables.path.find_db_qualified(thd, name,
+                                                 pkg_routine_handler, pkgname);
+    else
+    {
+      if ((*pkg_routine_handler)->sp_find_routine_quick(thd, name))
```

Comment:
I don't understand. is db name part of name? If not, sp_find_routine_quick, likely, won't work. So it is, right? Where it's getting db name from?

Reply:
I mean, if there is a current database, then db name could be a current database. But here you're in the
```c
if (!thd->db.length)
```
branch, there is no current database. And `name->m_explicit_name` is also false. So db name can only be set if you've already searched the path earlier. Is it what happening here, db name is set from you searching the path?